pkg/controller/controlplane/utils_test.go

```diff
@@ -0,0 +1,122 @@
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	namespace = "default"
+)
+
+func TestController(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Controlplane Suite")
+}
+
+var _ = Describe("Utils", func() {
+	var (
+		ctrl *gomock.Controller
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+	})
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#DNSNamesForService", func() {
+		It("should return all expected DNS names for the given service name and namespace", func() {
+			Expect(DNSNamesForService("test", namespace)).To(Equal([]string{
+				"test",
+				"test." + namespace,
+				"test." + namespace + ".svc",
+				"test." + namespace + ".svc.cluster.local",
+			}))
+		})
+	})
+
+	Describe("#ComputeCheckums", func() {
+		var (
+			secrets = map[string]*corev1.Secret{
+				"test-secret": getSecret(namespace, "test-secret", map[string][]byte{"foo": []byte("bar")}),
+			}
+			cms = map[string]*corev1.ConfigMap{
+				"test-config": getConfigMap(namespace, "test-config", map[string]string{"abc": "xyz"}),
+			}
+			existingSecret = getSecret(namespace, "existing-secret", map[string][]byte{"123": []byte("456")})
+			existingConfig = getConfigMap(namespace, "existing-config", map[string]string{"right": "wrong"})
+		)
+		It("should compute all checksums for the given secrets and configmpas", func() {
+
+			// Create mock client
+			c := mockclient.NewMockClient(ctrl)
+			c.EXPECT().Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: "existing-secret"}, &corev1.Secret{}).DoAndReturn(
+				func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+					*obj.(*corev1.Secret) = *existingSecret
+					return nil
+				},
+			)
+			c.EXPECT().Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: "existing-config"}, &corev1.ConfigMap{}).DoAndReturn(
+				func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+					*obj.(*corev1.ConfigMap) = *existingConfig
+					return nil
+				},
+			)
+
+			checksums, err := ComputeChecksums(context.TODO(), c, secrets, cms, []string{"existing-secret"}, []string{"existing-config"}, namespace)
+			Expect(err).To(Not(HaveOccurred()))
+			Expect(checksums).To(Equal(map[string]string{
+				"test-secret":     "8bafb35ff1ac60275d62e1cbd495aceb511fb354f74a20f7d06ecb48b3a68432",
+				"test-config":     "08a7bc7fe8f59b055f173145e211760a83f02cf89635cef26ebb351378635606",
+				"existing-secret": "cc5a179ccb1ea04bbf07089102398ca852fc6ac4e5c076ab4451e191102b240f",
+				"existing-config": "66e322807a642afda73ea4696ff94362a3b402fcdbe4c5676543478c63f26bb9",
+			}))
+		})
+	})
+})
+
+func getSecret(namespace, name string, data map[string][]byte) *corev1.Secret {
```

Comment:
`getSecret` -> `mkSecret`

Reply:
Why? `mkSomething` is somewhat awkward naming in my opinion. It's not Go stylish, and I don't think it's used in either `gardener` or `gardener-extensions`. In general, if you suggest naming changes like this, could you point why do you think they should be made?

Reply:
This is Go and not Java. We are a community Go project, thus we should also strive to have code similar to the community for everyone to easily start working on it. `get` always indicates a more or less 'remote' call to retrieve something from somewhere. In this case, you're just building a `Secret`, thus `get` is a bad name.